### PR TITLE
docs: flesh out docker-compose documentation

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc,

Please review the following commits I made in branch nicks/ch1811/dcdocs:

ceac41818009dd225ed29f8f4c376fb8001253b7 (2019-03-13 13:28:12 -0400)
docs: flesh out docker-compose documentation

### CODE REVIEW

File: docs/docker_compose.md [hunk 1]
```diff
@@ -53,9 +60,106 @@ When you declare either `docker_build` or `fast_build`,
 Tilt will find the image name in your `docker-compose.yml`,
 and use its own container-updating strategy instead of the one in the `docker-compose.yml` file.
 
+## Sample Project With `docker_build`
+
+Let's look at a simple example app that runs Redis and a NodeJS-based server with docker-compose.
```
Comment (maiamcc) [inline docs/docker_compose.md]: nit: `docker-compose` is the command. the tool is called `Docker Compose`.

File: docs/docker_compose.md [hunk 2]
```diff
@@ -36,7 +36,14 @@ That's it! Then run:
...
+Tilt will pick up your Docker Compose file and start running your services.
+
+Be aware of one important difference between `tilt up` and `docker-compose up`: Tilt
+will leave your services up when it exists. To turn the services down, run:
+
+```
+$ tilt down
+```
 
 ## Making Changes
```
Comment (maiamcc) [inline docs/docker_compose.md]: a little preamble here to make it clear that this is optional/an optimization? something to the effect of "you CAN just let `docker compose up` take care of building the images as well as running them. However, if you want to leverage Tilt's image-building expertise, you can let us build your images for you..."

File: docs/docker_compose.md [hunk 3]
```diff
@@ -53,9 +60,106 @@ When you declare either `docker_build` or `fast_build`,
...
+same project as above.
+
+First, we want to remove all the lines from the Dockerfile that add source code. Now our Dockerfile is:
+
+```dockerfile
+FROM node:9-alpine
+WORKDIR '/var/www/app'
+```
+
+Now, we put those lines for installing dependencies and adding source into our Tiltfile:
```
Comment (maiamcc) [inline docs/docker_compose.md]:
```suggestion
Now, we put those lines for installing dependencies and adding source code into our Tiltfile:
```

File: docs/docker_compose.md [hunk 4]
```diff
@@ -53,9 +60,106 @@ When you declare either `docker_build` or `fast_build`,
...
+docker_compose('docker-compose.yml')
+
+img = fast_build(
+  'tilt.dev/express-redis-app',
+  'Dockerfile',
+  'node server.js')
+
+img.add('.', '/var/www/app')
+
+img.run('npm i', trigger='package.json')
```
Comment (maiamcc) [inline docs/docker_compose.md]:
```suggestion
img.run('npm install', trigger='package.json')
```
(more obvious what's going on)

File: docs/docker_compose.md (hunk 1 above)
Comment (nicks) [reply docs/docker_compose.md]: good point!

File: docs/docker_compose.md (hunk 2 above)
Comment (nicks) [reply docs/docker_compose.md]: i like that! tweaked a bit and added

File: docs/docker_compose.md (hunk 4 above)
Comment (nicks) [reply docs/docker_compose.md]: done

File: docs/docker_compose.md (hunk 3 above)
Comment (nicks) [reply docs/docker_compose.md]: done

**Pull request merged**